python/cudf/cudf/core/frame.py

```diff
@@ -140,11 +140,19 @@ def _slice(self, arg):
         else:
             start = len(self) if start > num_rows else start
             stop = len(self) if stop > num_rows else stop
+            from cudf.core.index import RangeIndex
 
-            result = self._from_table(
-                libcudfxx.copying.table_slice(self, [start, stop])[0]
-            )
-
+            if isinstance(self._index, RangeIndex):
+                new_index = self._index[start:stop]
+                self = self.reset_index(drop=True)
+                result = self._from_table(
+                    libcudfxx.copying.table_slice(self, [start, stop])[0]
+                )
```

Comment:
This will just materialize the new RangeIndex generated from the `reset_index` call. We need to modify `table_slice` to have the ability to ignore the index.